tests/ProtocolBlockTypePalette/ProtocolBlockTypePaletteTest.cpp

```diff
@@ -0,0 +1,173 @@
+// ProtocolBlockTypePaletteTest.cpp
+
+#include <string>
+#include <fstream>
+#include <streambuf>
+#include "Globals.h"
+
+#include "Protocol/ProtocolBlockTypePalette.h"
+#include "json/json.h"
+
+#include "../TestHelpers.h"
+
+
+
+
+
+static void TestInit(void)
+{
+	LOG("Test TestInit");
+	ProtocolBlockTypePalette palette("{}");
+}
+
+
+
+
+
+static void TestSuccess(void)
+{
+	LOG("Test TestSuccess");
+	ProtocolBlockTypePalette palette;
+	TEST_TRUE(palette.loadFromString("[]"));
+
+	auto example = "{\"Metadata\":{\"ProtocolBlockType\":1}, \"Palette\":[{\
+		\"props\": {\
+			\"foo\": \"bar\"\
+		}, \
+		\"name\": \"b\", \
+		\"id\": \"0\"\
+	}]}";
+
+	palette.clear();
+	TEST_TRUE(palette.loadFromString(example));
+	TEST_EQUAL(palette.index("b", BlockState({{"foo", "bar"}})), 0);
+	TEST_EQUAL(palette.index("b", BlockState({{"foo", "baz"}})), ProtocolBlockTypePalette::NOT_FOUND);
+	TEST_EQUAL(palette.index("a", BlockState({{"foo", "bar"}})), ProtocolBlockTypePalette::NOT_FOUND);
+}
+
+
+
+
+
+static void TestErrors(void)
+{
+	LOG("Test TestErrors");
+	ProtocolBlockTypePalette palette;
+	TEST_THROWS_ANY(palette.loadFromString(""));
```

Comment:
You wrote that throwing would be fine... I wrote that it **can** throw, not that it throws instead of returning `false`.

The exception comes from the JSON parser, the message should be something about `object or array expected` for the empty one.

I guess I'll go back to catching exceptions from the JSON parser within the class...